File: GGAndroid_2.0/GGDemo/src/main/java/io/ganguo/demo/view/activity/CoordinatorAppBarDemoActivity.java
package io.ganguo.demo.view.activity;

import android.content.Context;
import android.content.Intent;

import io.ganguo.demo.view.activity.base.GGVModelActivity;
import io.ganguo.demo.viewmodel.activity.CoordinatorAppBarDemoVModel;
import io.ganguo.vmodel.ViewModelActivity;
import io.ganguo.viewmodel.databinding.IncludeCoordinatorLayoutAppBarBinding;
import io.ganguo.vmodel.ViewModelHelper;

/**
 * CoordinatorAppBarVModel Demo
 * Created by leo on 2018/7/13.
 */
public class CoordinatorAppBarDemoActivity extends GGVModelActivity<IncludeCoordinatorLayoutAppBarBinding, CoordinatorAppBarDemoVModel> {

    public static Intent intentFor(Context context) {
        return new Intent(context, CoordinatorAppBarDemoActivity.class);
    }

    @Override
    public CoordinatorAppBarDemoVModel createViewModel() {
        return new CoordinatorAppBarDemoVModel();
    }

    @Override
    public void onViewAttached(CoordinatorAppBarDemoVModel viewModel) {
        ViewModelHelper.bind(viewModel.getHeaderContainer(), viewModel, onCreateHeaderVModel());
    }
}

File: GGAndroid_3.0/main/app-mvvm-v1-sample/src/main/java/io/ganguo/demo/view/activity/CoordinatorAppBarDemoActivity.kt
package io.ganguo.demo.view.activity

import android.content.Context
import android.content.Intent
import io.ganguo.demo.view.activity.base.GGVModelActivity
import io.ganguo.demo.viewmodel.activity.CoordinatorAppBarDemoVModel
import io.ganguo.viewmodel.core.BaseViewModel
import io.ganguo.viewmodel.pack.common.HeaderViewModel
import io.ganguo.viewmodel.databinding.IncludeCoordinatorLayoutAppBarBinding
import io.ganguo.viewmodel.core.ViewModelHelper

/**
 * CoordinatorAppBarVModel Demo
 * Created by leo on 2018/7/13.
 */
class CoordinatorAppBarDemoActivity : GGVModelActivity<IncludeCoordinatorLayoutAppBarBinding, CoordinatorAppBarDemoVModel>() {

    override fun createViewModel(): CoordinatorAppBarDemoVModel {
        return CoordinatorAppBarDemoVModel()
    }

    override fun onViewAttached(viewModel: CoordinatorAppBarDemoVModel) {
        ViewModelHelper.bind(viewModel.headerContainer, viewModel, newHeaderVModel())
    }

    companion object {

        fun intentFor(context: Context): Intent {
            return Intent(context, CoordinatorAppBarDemoActivity::class.java)
        }
    }

}
